tests/test_frontend.py

```diff
@@ -24,3 +24,11 @@ def test_add_contact_to_contacts(self):
         driver.find_element_by_id("submit").click()
 
         assert ("Kota" in driver.page_source)
+
+    def test_delete_contact(self):
+        driver = webdriver.Chrome("/usr/local/bin/chromedriver")
+        driver.set_page_load_timeout(10)
+        driver.get("http://127.0.0.1:5000/contacts")
+        driver.find_element_by_id("delete").click()
+
+        assert not ("Kota" in driver.page_source)
```

Comment:
It looks like this test isn't actually verifying the removal of a contact. Or I'm missing something here.

But where is the person `Kota` added here? Looks like it's "passing" because that sub string was never part of the page source in the first place.

Reply:
Oh I think I got it now. This test relies on the other test run first, doesn't it? The _other_ test adds a contact, which this test expects to be there.

🤔 

Can you think of a way to have each test set up "the world" they need. So that you could randomise them or even run them individually?

Reply:
Hi @christophgockel , you're right, it's passing because of the previous test. The same applies to counting (`len`) from 1 not 0 - it's because one contact was already added.

I think I can reopen the server after each test so that each time there will be a new environment. What do you think about it?

Reply:
I've decided to delete old contacts before running new test, so that I won't have to restart the server.